bookkeeper-server/src/test/java/org/apache/bookkeeper/replication/AuditorPeriodicCheckTest.java

```diff
@@ -454,7 +457,100 @@ void validateInitialDelayOfCheckAllLedgers(LedgerUnderreplicationManager urm, lo
+        validateInitialDelayOfMetadataCheck(urm, 999, 1000, servConf, bkc);
+        validateInitialDelayOfMetadataCheck(urm, 1001, 1000, servConf, bkc);
+    }
+
+    void validateInitialDelayOfMetadataCheck(LedgerUnderreplicationManager urm, long timeSinceLastExecutedInSecs,
+            long auditorPeriodicMetadataCheckInterval, ServerConfiguration servConf, BookKeeper bkc)
+            throws UnavailableException, UnknownHostException, InterruptedException {
+        TestStatsProvider statsProvider = new TestStatsProvider();
+        TestStatsLogger statsLogger = statsProvider.getStatsLogger(AUDITOR_SCOPE);
+        TestOpStatsLogger metadataCheckStatsLogger = (TestOpStatsLogger) statsLogger
+                .getOpStatsLogger(ReplicationStats.METADATA_CHECK_TIME);
+        servConf.setAuditorPeriodicMetadataCheckInterval(auditorPeriodicMetadataCheckInterval);
+        servConf.setAuditorPeriodicCheckInterval(0);
+        servConf.setAuditorPeriodicBookieCheckInterval(0);
+        final TestAuditor auditor = new TestAuditor(Bookie.getBookieAddress(servConf).toString(), servConf, bkc, false,
+                statsLogger);
+        CountDownLatch latch = auditor.getLatch();
+        assertEquals("METADATA_CHECK_TIME SuccessCount", 0, metadataCheckStatsLogger.getSuccessCount());
+        long curTimeBeforeStart = System.currentTimeMillis();
+        long metadataCheckCTime = -1;
+        long initialDelayInMsecs = -1;
+        long nextExpectedMetadataCheckExecutionTime = -1;
+        long bufferTimeInMsecs = 12000L;
+        if (timeSinceLastExecutedInSecs == -1) {
+            /*
+             * if we are setting metadataCheckCTime to -1, it means that
+             * metadataCheck hasn't run before. So initialDelay for
+             * metadataCheck should be 0.
+             */
+            metadataCheckCTime = -1;
+            initialDelayInMsecs = 0;
+        } else {
+            metadataCheckCTime = curTimeBeforeStart - timeSinceLastExecutedInSecs * 1000L;
+            initialDelayInMsecs = timeSinceLastExecutedInSecs > auditorPeriodicMetadataCheckInterval ? 0
+                    : (auditorPeriodicMetadataCheckInterval - timeSinceLastExecutedInSecs) * 1000L;
+        }
+        /*
+         * next metadataCheck should happen atleast after
+         * nextExpectedMetadataCheckExecutionTime.
+         */
+        nextExpectedMetadataCheckExecutionTime = curTimeBeforeStart + initialDelayInMsecs;
+
+        urm.setMetadataCheckCTime(metadataCheckCTime);
+        auditor.start();
+        /*
+         * since auditorPeriodicMetadataCheckInterval are higher values (in the
+         * order of 100s of seconds), its ok bufferTimeInMsecs to be ` 10 secs.
+         */
+        assertTrue("metadataCheck should have executed with initialDelay " + initialDelayInMsecs,
+                latch.await(initialDelayInMsecs + bufferTimeInMsecs, TimeUnit.MILLISECONDS));
+        for (int i = 0; i < 10; i++) {
```

Comment:
This would be a new flaky test on slow machines
Maybe it is better to have a longer timeout